crates/matrix-sdk-base/src/client.rs

```diff
@@ -1173,6 +1180,15 @@ impl BaseClient {
             }
         }
 
+        for (room_id, pinned_event_ids) in changes.pinned_events.clone() {
```

Comment:
nit: you shouldn't need the `clone()` here, and as said before `changes.pinned_events` should be a local variable

Reply:
Removed in https://github.com/matrix-org/matrix-rust-sdk/pull/3747/commits/8c43c7523b37a595cc9b6cddcf22da78d560c46b